### PR TITLE
Add API key validation

### PR DESCRIPTION
**Pull request by ketphan02**

### CODE REVIEW

File: api/api/views/generate_teams.py [hunk 1]
```diff
@@ -28,6 +29,15 @@ def generate_teams(self, request):
         6. Return JSON to a happy user :) 🚀!
         """
         try:
+            request_headers = dict(request.headers)
```
Comment (Opeyem1a) [inline api/api/views/generate_teams.py:32]: I feel like this should be middleware, not just in this view

File: utils/api_auth.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import os
+from bcrypt import hashpw, gensalt, checkpw
+
+from pathlib import Path
+import dotenv
+
+env_file = next((Path(__file__).parent.parent / "env").glob("*.env"))
```
Comment (Opeyem1a) [inline utils/api_auth.py:7]: what does this do?

File: utils/api_auth.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+    This function only serves to encrypt a new API token for customers and should never get called in production.
+    """
+    return hashpw(
+        password=api_token.encode("utf-8"),
+        salt=gensalt(),
+    )
+
+
+def check_hash(api_token, encrypted_api_token):
```
Comment (Opeyem1a) [inline utils/api_auth.py:23]: nit; 

does this return bool? generally why no types

File: api/api/settings/__init__.py [hunk 4]
```diff
@@ -29,7 +25,7 @@
 # SECURITY WARNING: don't run with debug turned on in production!
 DEBUG = True
 
-ALLOWED_HOSTS = list(os.environ.get("DJANGO_ALLOWED_HOSTS", "").split(" "))
+ALLOWED_HOSTS = ["*"]
```
Comment (Opeyem1a) [inline api/api/settings/__init__.py:28]: why?

**Review by Opeyem1a — CHANGES_REQUESTED** [review]

File: api/api/settings/__init__.py (hunk 4 above)
Comment (ketphan02) [reply api/api/settings/__init__.py:28]: This is dev, we still have this on prod. This makes the API work locally even without an `.env`

File: utils/api_auth.py (hunk 2 above)
Comment (ketphan02) [reply utils/api_auth.py:7]: This loads either `.env` or `production.env`  inside of the `env` folder (I adopted this structure from Gamification)

**Review by Opeyem1a — APPROVED** [review]

**Pull request merged**